pkgs/mime/lib/src/mime_type.dart

```diff
@@ -64,19 +52,39 @@ class MimeTypeResolver {
       }
     }
     final ext = _ext(path);
-    result = _extensionMap[ext];
+    result = _extensionToMime[ext];
     if (result != null) return result;
     if (_useDefault) {
-      result = defaultExtensionMap[ext];
+      result = extensionToMime[ext];
       if (result != null) return result;
     }
     return null;
   }
 
+  /// The default file extension for a given MIME type.
+  ///
+  /// If [mimeType] has multiple associated extensions, the returned string is
+  /// one of those, chosen as the default extension for that MIME type.
+  ///
+  /// Returns `null` if [mimeType] is not a recognized and supported MIME type.
+  String? extensionFromMime(String mimeType) {
+    if (_useDefault) {
+      return mimeToExtensions[mimeType.toLowerCase()]?.first;
+    }
+
+    for (final entry in _extensionToMime.entries) {
+      if (entry.value == mimeType) {
+        return entry.key;
```

Comment:
This returns the first entry with that media type as value.
Is that guaranteed to be the canonical one? (Probably, if it's a linked hash map and the canonical one was added first. But might be worth mentioning.)

Reply:
I'm certain I saw wording to that effect in one of the mime registries I was looking at, but don't see that in the docs for the apache httpd mime list. Their mime registry may not have an idea of a canonical file ext. :(

Reply:
That would be the android one: https://android.googlesource.com/platform/frameworks/base/+/769f04c8f03a/mime/java-res/android.mime.types

"defines a mapping from one MIME type to the first of the listed extensions,"